include/deal.II/lac/la_vector.h

```diff
@@ -214,7 +214,7 @@ namespace LinearAlgebra
      *  vec.locally_owned_elements() == complete_index_set(vec.size())
      * @endcode
      */
-    const dealii::IndexSet &locally_owned_elements() const;
```

Comment:
What is the reasoning behind this change? You don't want every vector to keep an IndexSet around? If you return by value, does it need to be const?